Dockerfile.uptermd

```diff
@@ -5,18 +5,21 @@ FROM golang:latest AS builder
 
 ARG TARGETOS
 ARG TARGETARCH
-ARG VERSION=0.0.0+dev
-ARG GIT_COMMIT=unknown
-ARG BUILD_DATE=unknown
 
 WORKDIR /src
 ENV CGO_ENABLED=0
-RUN --mount=target=. \
-  --mount=type=cache,target=/root/.cache/go-build \
+ENV GOOS=$TARGETOS
+ENV GOARCH=$TARGETARCH
+ARG GORELEASER_VERSION=2.12.7
+ARG GORELEASER_ARCH=amd64
+
+RUN wget https://github.com/goreleaser/goreleaser/releases/download/v${GORELEASER_VERSION}/goreleaser_${GORELEASER_VERSION}_${GORELEASER_ARCH}.deb && \
+    dpkg -i goreleaser*.deb
+
+COPY . .
+RUN --mount=type=cache,target=/root/.cache/go-build \
   --mount=type=cache,target=/go/pkg \
-  GOOS=$TARGETOS GOARCH=$TARGETARCH go install \
-  -ldflags="-s -w -X github.com/owenthereal/upterm/internal/version.Version=${VERSION} -X github.com/owenthereal/upterm/internal/version.GitCommit=${GIT_COMMIT} -X github.com/owenthereal/upterm/internal/version.Date=${BUILD_DATE}" \
-  ./cmd/...
+  goreleaser build --single-target --auto-snapshot
```

Comment:
Note: `--auto-snapshot` automatically sets `--snapshot` if the repository is dirty, this happens if:
- files are modified
- untracked files are present

Here's what versions look like `(docker build -f Dockerfile.uptermd -t uptermd . && docker run --rm uptermd version`)

Repo is clean (no modifications, no untracked files): 
```
Uptermd version 0.19.0
Git commit: 522ecd5085d8998d72bfa3f63c04fb53e5cbcd77
Build date: 2025-11-30T13:29:39Z
```

Repo is dirty (untracked files are present):

```
Uptermd version 0.19.1-snapshot
Git commit: 522ecd5085d8998d72bfa3f63c04fb53e5cbcd77
Build date: 2025-11-30T13:29:39Z
```